lib/internal/Magento/Framework/Data/Collection.php

```diff
@@ -611,6 +616,13 @@ public function setItemObjectClass($className)
         if (!is_a($className, DataObject::class, true)) {
             throw new \InvalidArgumentException($className . ' does not extend \Magento\Framework\DataObject');
         }
+        
+        if ($this->newEmptyItem !== null
+            && $this->_itemObjectClass !== $className
+        ) {
+            $this->newEmptyItem === null;
+        }
```

Comment:
I think it's redundant code because this is mean that one collection could have multiple items with different classes.

Reply:
@Den4ik  I've removed the redundant code.